src/OpenApi/Generate.elm

```diff
@@ -1878,6 +1878,39 @@ typeToDecoder type_ =
             CliMonad.todo "Bytes decoder not implemented yet"
 
 
+jsonDecodeAndMap : Elm.Declaration
+jsonDecodeAndMap =
```

Comment:
I tried using `Elm.Declaration.fn2` but the type annotation kept being generated as
```elm
Json.Decode.Decoder a -> Json.Decode.Decoder (a -> b) -> Json.Decode.Decoder value
```
when I used `Elm.val "(|>)"`, and when I tried used an anonymous function instead running the codegen would hang indefinitely.

This felt like a fine compromise

Reply:
It's fine. Eventually we'll have @janiczek's type inference and all these bugs will be lost in time, like tears in rain...